compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts

```diff
@@ -159,7 +178,23 @@ test_cases.forEach((testInfo) => {
           isRecursive,
         );
 
-        expect(verified).to.be.true;
+        expect(verified, "Proof fails verification in JS").to.be.true;
+
+        try {
+          let result;
+          if (testInfo.publicInputsLength === 0) {
+            result = await contract.verify(proof, []);
+          } else {
+            const publicInputs = proof.slice(0, testInfo.publicInputsLength);
+            const slicedProof = proof.slice(testInfo.publicInputsLength);
+            result = await contract.verify(slicedProof, publicInputs);
```

Comment:
We don't actually have a test for this codepath. Can you add a new circuit with a public input? We have one [here](https://github.com/noir-lang/noir/tree/master/compiler/integration-tests/test/circuits/main) which is suitable.